src/main/scala/me/rexim/morganey/Commands.scala

```diff
@@ -9,24 +9,31 @@ import scala.util.{Failure, Success}
 
 object Commands {
 
-  final type Command = ReplContext => Computation[ReplResult[String]]
+  type CommandTask = ReplContext => Computation[ReplResult[String]]
 
-  private def command(f: ReplContext => ReplResult[String]): Command =
+  case class Command(
+    name: String,
+    task: String => ReplContext => ReplResult[String],
+    hasTermArgument: Boolean)
```

Comment:
Can we get rid of that? I mean it's no better than having a function with a boolean argument. Why can't it be something like

```scala
trait Command
case class StringCommand extends Command
case class TermCommand extends Command
```